### PR TITLE
[Effects] Gfx data + stars_burst

### PR DESCRIPTION
**Pull request by mrshigure**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀 This PR matches 3 functions (+0.05%) on `us`.

File: src/effects/gfx/ending_decals.c [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
..., 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0
```
Comment (nanaian) [inline src/effects/gfx/ending_decals.c:10]: Tbh might be better to `u8 D_09001C20_36CB00[explicit length] = {};` and let this zero-initialize on its own

Comment (nanaian) [inline src/effects/gfx/ending_decals.c:10]: Tbh might be better to do `u8 D_09001C20_36CB00[explicit length] = {};` and let these zero-initialize on their own?

File: src/effects/gfx/drop_leaves.c [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#include "effects/gfx/D_09001000_33E610.vtx.inc.c"
+#include "effects/gfx/D_09001080_33E690.vtx.inc.c"
+#include "effects/gfx/D_09001100_33E710.vtx.inc.c"
+
+#include "effects/gfx/D_09001180_33E790.gfx.inc.c"
+#include "effects/gfx/D_09001230_33E840.gfx.inc.c"
+#include "effects/gfx/D_09001258_33E868.gfx.inc.c"
+#include "effects/gfx/D_09001280_33E890.gfx.inc.c"
+
+u8 D_090012A8_33E8B8[] = {
```
Comment (ethteck) [inline src/effects/gfx/drop_leaves.c:14]: these end padding syms can actually be removed

File: src/effects/gfx/effect_3D.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#include "common.h"
+
+#include "effects/gfx/D_09000000_37D180.png.inc.c"
+
+#include "effects/gfx/D_09000200_37D380.vtx.inc.c"
+
+#include "effects/gfx/D_09000240_37D3C0.gfx.inc.c"
+#include "effects/gfx/D_090002E8_37D468.gfx.inc.c"
+
+u8 D_09000308_37D488[] = {
```
Comment (ethteck) [inline src/effects/gfx/effect_3D.c:10]: here2

File: src/effects/gfx/ending_decals.c [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#include "effects/gfx/D_09001EA8_36CD88.gfx.inc.c"
+#include "effects/gfx/D_09001F10_36CDF0.gfx.inc.c"
+#include "effects/gfx/D_09001F78_36CE58.gfx.inc.c"
+
+#include "effects/gfx/D_09002018_36CEF8.vtx.inc.c"
+
+#include "effects/gfx/D_09002058_36CF38.gfx.inc.c"
+#include "effects/gfx/D_09002078_36CF58.gfx.inc.c"
+
+u8 D_09002138_36D018[] = {
```
Comment (ethteck) [inline src/effects/gfx/ending_decals.c:25]: here2

File: src/effects/gfx/energy_in_out.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+#include "effects/gfx/D_09000748_3D6678.vtx.inc.c"
+#include "effects/gfx/D_09000788_3D66B8.vtx.inc.c"
+#include "effects/gfx/D_090007C8_3D66F8.vtx.inc.c"
+
+#include "effects/gfx/D_09000808_3D6738.gfx.inc.c"
+#include "effects/gfx/D_09000828_3D6758.gfx.inc.c"
+#include "effects/gfx/D_09000848_3D6778.gfx.inc.c"
+#include "effects/gfx/D_09000868_3D6798.gfx.inc.c"
+
+u8 D_09000888_3D67B8[] = {
```
Comment (ethteck) [inline src/effects/gfx/energy_in_out.c:47]: here2

File: src/effects/gfx/explosion.c [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+#include "effects/gfx/D_09000800_344770.vtx.inc.c"
+
+#include "effects/gfx/D_09000840_3447B0.gfx.inc.c"
+#include "effects/gfx/D_090008F0_344860.gfx.inc.c"
+#include "effects/gfx/D_09000910_344880.gfx.inc.c"
+#include "effects/gfx/D_09000978_3448E8.gfx.inc.c"
+#include "effects/gfx/D_090009D8_344948.gfx.inc.c"
+#include "effects/gfx/D_09000A38_3449A8.gfx.inc.c"
+
+u8 D_09000A98_344A08[] = {
```
Comment (ethteck) [inline src/effects/gfx/explosion.c:17]: here2

File: src/effects/gfx/falling_leaves.c [hunk 7]
```diff
@@ -0,0 +1,12 @@
+#include "common.h"
+
+#include "effects/gfx/D_09000000_35F0E0.png.inc.c"
+
+#include "effects/gfx/D_09001000_3600E0.vtx.inc.c"
+
+#include "effects/gfx/D_09001100_3601E0.gfx.inc.c"
+#include "effects/gfx/D_090011B0_360290.gfx.inc.c"
+
+u8 D_090011D8_3602B8[] = {
```
Comment (ethteck) [inline src/effects/gfx/falling_leaves.c:10]: here2

File: src/effects/gfx/fire_flower.c [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+#include "effects/gfx/D_09000CC0_381060.vtx.inc.c"
+#include "effects/gfx/D_09000D00_3810A0.vtx.inc.c"
+
+#include "effects/gfx/D_09000D40_3810E0.gfx.inc.c"
+#include "effects/gfx/D_09000DE0_381180.gfx.inc.c"
+#include "effects/gfx/D_09000EB8_381258.gfx.inc.c"
+#include "effects/gfx/D_09000ED8_381278.gfx.inc.c"
+#include "effects/gfx/D_09000EF8_381298.gfx.inc.c"
+
+u8 D_09000F18_3812B8[] = {
```
Comment (ethteck) [inline src/effects/gfx/fire_flower.c:30]: here2

File: src/effects/gfx/gather_magic.c [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+#include "effects/gfx/D_09000080_38F790.gfx.inc.c"
+
+#include "effects/gfx/D_09000128_38F838.vtx.inc.c"
+#include "effects/gfx/D_09000168_38F878.vtx.inc.c"
+
+#include "effects/gfx/D_090001A8_38F8B8.gfx.inc.c"
+#include "effects/gfx/D_090001C8_38F8D8.gfx.inc.c"
+
+u8 D_090001E8_38F8F8[] = {
```
Comment (ethteck) [inline src/effects/gfx/gather_magic.c:13]: here2

File: src/effects/gfx/got_item_outline.c [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+#include "effects/gfx/D_09008000_34DB40.vtx.inc.c"
+
+#include "effects/gfx/D_09008100_34DC40.gfx.inc.c"
+#include "effects/gfx/D_09008170_34DCB0.gfx.inc.c"
+#include "effects/gfx/D_09008190_34DCD0.gfx.inc.c"
+#include "effects/gfx/D_090081A8_34DCE8.gfx.inc.c"
+#include "effects/gfx/D_090081C0_34DD00.gfx.inc.c"
+
+u8 D_090081D8_34DD18[] = {
```
Comment (ethteck) [inline src/effects/gfx/got_item_outline.c:20]: here2

File: src/effects/gfx/lil_oink.c [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+#include "effects/gfx/D_09003228_402478.gfx.inc.c"
+#include "effects/gfx/D_090032C0_402510.gfx.inc.c"
+#include "effects/gfx/D_09003358_4025A8.gfx.inc.c"
+#include "effects/gfx/D_09003370_4025C0.gfx.inc.c"
+#include "effects/gfx/D_09003388_4025D8.gfx.inc.c"
+#include "effects/gfx/D_090033A0_4025F0.gfx.inc.c"
+#include "effects/gfx/D_090033B8_402608.gfx.inc.c"
+#include "effects/gfx/D_090033D0_402620.gfx.inc.c"
+
+u8 D_090033E8_402638[] = {
```
Comment (ethteck) [inline src/effects/gfx/lil_oink.c:149]: here2

File: ver/us/splat.yaml [hunk 12]
```diff
@@ -3202,7 +3454,33 @@ segments:
...
+      - [0x36CB00] # padding
+      - [0x36CCE0, vtx, D_09001E00_36CCE0]
+      - [0x36CD20, gfx, D_09001E40_36CD20]
+      - [0x36CD88, gfx, D_09001EA8_36CD88]
+      - [0x36CDF0, gfx, D_09001F10_36CDF0]
+      - [0x36CE58, gfx, D_09001F78_36CE58]
+      - [0x36CEF8, vtx, D_09002018_36CEF8]
+      - [0x36CF38, gfx, D_09002058_36CF38]
+      - [0x36CF58, gfx, D_09002078_36CF58]
+      - [0x36D018] # padding
```
Comment (ethteck) [inline ver/us/splat.yaml:3483]: these end padding comments can be removed (I think the ones mid-subsegments are fine though)

**Review by ethteck — CHANGES_REQUESTED** [review]
omg great stuff :) so many effects graphics dun

File: ver/us/splat.yaml (hunk 12 above)
Comment (mrshigure) [reply ver/us/splat.yaml:3483]: Do you mean removing the comment only (`# padding`) or the whole subsegment (`- [0xXXXXXX] #padding`)? Removing the whole thing seems to work without messing the previous subsegment but I'm not sure why.

Comment (ethteck) [reply ver/us/splat.yaml:3483]: I mean to just remove the comments. it's still good to note the data ends at this spot

File: src/effects/gfx/ending_decals.c (hunk 1 above)
Comment (mrshigure) [reply src/effects/gfx/ending_decals.c]: Updated. Way cleaner now.

**Pull request merged**